### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by Andruschenko** [razor1895/react-native-refreshable-flatlist#1]
I'm experiencing some performance errors in the console of the react-native debugger [of the kind '[Violation] 'message' handler took XXXms'](https://stackoverflow.com/questions/41218507/violation-long-running-javascript-task-took-xx-ms) 
where XXX gets up to 767ms.

The FlatList I am using contains many images. The messages occur after I release at the bottom, i.e. when the `onRelease` callback is called. 

Does anyone else experience these problems?

**Comment by razor1895** [comment]
could you please give a demo repo that can reproduce this issue?

**Comment by stokesbga** [comment]
I am also experiencing serious issues with performance. Will see if there is an improvement on release. Im on 0.43

*EDIT* I swapped out the regular indicator for the custom indicator, and added `useNativeDriver: true` to the timing statements, and there is a drastic improvement. The layout is still all over the place though. Maybe add to readme how to recreate the demo in your gif?

**Comment by stokesbga** [comment]
Also, I dont know about others, but I wanted to use this library when I couldnt find a resource that allowed custom pull distance for the regular FlatList. I had to pull with several gestures to trigger the onRefresh.

**Comment by razor1895** [comment]
@stokesbga Thanks for your reply and I will provide a example project about how to create custom indicator

**Comment by razor1895** [comment]
https://gist.github.com/razor1895/2e755135af960752c164fad62f169ec9 here's the code will shows the same demo in the gif

**Comment by anil1712** [comment]
Yes, I also noticed major performance issue when I try to wrap this with [native-base](https://docs.nativebase.io/Components.html#Components) components. List are not getting rendered properly
